VoxelSniperCore/src/main/java/com/github/kevindagame/command/VoxelPerformerCommand.java

```diff
@@ -37,31 +37,36 @@ public boolean doCommand(IPlayer player, String[] args) {
             return true;
         }
 
-
+        IBrush brush = sniper.getBrush(sniper.getCurrentToolId());
         if (args.length == 0) {
-            IBrush brush = sniper.getBrush(sniper.getCurrentToolId());
-            if (brush instanceof IPerformerBrush) {
-                ((IPerformerBrush) brush).parsePerformer("m", snipeData);
-            } else {
-                sniper.sendMessage(Messages.THE_ACTIVE_BRUSH_IS_NOT_A_PERFORMER_BRUSH);
-            }
-            return true;
+            return setPerformer(brush, "m", sniper);
         }
 
         if (args.length == 1) {
-            IBrush brush = sniper.getBrush(sniper.getCurrentToolId());
-            if (brush instanceof IPerformerBrush) {
-                boolean success = ((IPerformerBrush) brush).parsePerformer(args[0], snipeData);
-                if (!success) {
-                    sniper.sendMessage(Messages.NO_SUCH_PERFORMER.replace("%arg%", args[0]));
+            return setPerformer(brush, args[0], sniper);
+        }
+        return false;
+    }
+
+    private boolean setPerformer(IBrush brush, String performer, Sniper sniper) {
+        if (brush instanceof IPerformerBrush) {
+            return ((IPerformerBrush) brush).parsePerformer(performer, sniper.getSnipeData(sniper.getCurrentToolId()));
+        } else if (brush instanceof PolyBrush) {
+            var properties = ((PolyBrush) brush).getProperties();
+            for (var property : properties) {
+                if (property instanceof PerformerProperty) {
+                    (property).set(performer);
+                    brush.info(sniper.getSnipeData(sniper.getCurrentToolId()).getVoxelMessage());
+                    return true;
                 }
-            } else {
-                sniper.sendMessage(Messages.THE_ACTIVE_BRUSH_IS_NOT_A_PERFORMER_BRUSH);
             }
-            return true;
+            sniper.sendMessage(Messages.THE_ACTIVE_BRUSH_IS_NOT_A_PERFORMER_BRUSH);
+            return false;
```

Comment:
Couldn't we make PolyBrush implement IPerformerBrush and put this logic in the Polybrush? Might be nicer that way.

Reply:
No we can't. The flow of a performer brush vs a poly brush is different. Not all polybrushes use performers

Reply:
Makes sense now that I think of it.